sched/scheduler/jobRunner_test.go

```diff
@@ -40,8 +36,7 @@ func Test_RunTask_PanicWhenWritingStartTaskReturnsFatalError(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
-	// ensures node does not execute task when a fatal error occurs logging starttask
-	nodeMock := cm.NewMockNode(mockCtrl)
+	worker := worker.NewNoopWorker()
```

Comment:
With the way it was written a mock will cause a failure?  How will the fake do this?